### PR TITLE
Use lint checker

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This basically runs "android lint" and compares the error/warnings counts against a threshold in scripts/lint/lint-results.txt.
It will fail if the count increases and success if it is unchanged or decreased. 
If it decreases the new value will be pushed to the current source branch, triggering a new build.
(If we skip drone, then github would remain in "waiting state" and thus not allow us to merge it, so we have to start drone once again).

Main goal is to reduce the error/warning count to 0 or at least not introduce new issues.

@mario @AndyScherzinger

### CODE REVIEW

File: scripts/lint/lint-up.rb [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+system ("git config --local user.name '"  + git_user + "'")
+system ("git config --local user.email '.'") # set email blank
+system ("git remote rm origin")
+system ("git remote add origin https://" + git_user + ":" + git_token + "@github.com/nextcloud/android")
+
+# add previous Lint result file to git
+system ('git add ' + PREVIOUS_LINT_RESULTS_FILE)
+
+# Travis has git in a detached head here. Let's get on the right branch.
+travis_branch_name = ENV['TRAVIS_BRANCH']
```
Comment (AndyScherzinger) [inline scripts/lint/lint-up.rb]: @tobiasKaminsky We use travisci in some place?

**Comment by mario** [comment]
Is this ready for review?

**Comment by tobiasKaminsky** [comment]
ready to review

**Comment by AndyScherzinger** [comment]
👍 Not a ruby person and drone is still lid green, so it's fine by me

**Comment by mario** [comment]
I fixed a few issues. @AndyScherzinger - please re-approve.

**Comment by mario** [comment]
👍

**Comment by AndyScherzinger** [comment]
👍 re-approved

**Pull request merged**